### DISCUSSION TITLE
[Auto-publish] Metaphore to Hasnode ?

### DISCUSSION THREAD

**Discussion by mkubdev** [StreetCommunityProgrammer/metaphore#61] [Ideas]
## Introduction

What if we start using GitHub to publish articles on Hashnode?

## GitHub Application

There is an app, [ `Hashnode - Github As Source` ](https://github.com/apps/hashnode-github-as-source) which permits us to auto-publish `metaphores` on our Hashnode account. But we will need to update accordingly our Frontmatter headers to work.

This is [an example](https://github.com/Hashnode/Hashnode-source-from-github-template) of Frontmatter header addition:
```md
---
title: YOUR_TITLE_HERE **(Required)**
subtitle: YOUR_SUBTITLE_HERE
/* The pathname of your article url 
 * Ex. In https://engineering.hashnode.com/how-we-autodetect-spam-using-googles-vertex-ai 
   "how-we-autodetect-spam-using-googles-vertex-ai" is the slug */ 
slug: CUSTOM_ARTICLE_SLUG_HERE (Will be created automatically if not provided)
tags: TAG_SLUG_1, TAG_SLUG_2 **(Required)** - You can find the list of tags here https://github.com/Hashnode/support/blob/main/misc/tags.json
/* You need to upload your image to https://hashnode.com/uploader 
and use the uploaded image URL as COVER_IMAGE_URL */ 
cover: COVER_IMAGE_URL
domain: YOUR_HASHNODE_BLOG_DOMAIN_NAME_HERE (e.g. sandeep.dev or sandeep.hashnode.dev) **(Required)** 
/* When you have a team publication and a publication member has created an article */ 
/* Please note that this param is only supported while creating an article and not updating */ 
publishAs: USERNAME_OF_AUTHOR_OF_ARTICLE 
/* When your article is republished and you want to use  the origin url as canonical url */ 
canonical: CANONICAL_URL_OF_ARTICLE 
/* Only if you don't want this article to be published yet */ 
ignorePost: Boolean (default: false)
/* If you want to post it only on your blog and hide it from Hashnode community feed */
hideFromHashnodeCommunity: Boolean (default: false)
---
```
## Requirments

Actually, my Hashnode account is linked to GitHub.

We'll need to create:
1. New Gmail account
2. New Hashnode account from the mail

**Comment by darkterminal** [comment]
This idea is already mentioned on #28 

- We need to store API Key for each member in secure location.
- Check the API Key by Punk username if exists
- Distributed their content on their DEV/Hashnode/Medium account by GH Action and adding our signature on the bottom of their story

What we have right now?
- SCP have DEV organization account (I will invite you and other member to create other story)
- SCP have Twitter Account and we can use the account to tweet new story fro gh action.

What you think?

**Reply by mkubdev** [reply]
Oh, I forget #28 ! Yep, but why should we store members' API key?
The GHAction should take our SCP Hashnode API key for each action no?

I'm ok, we should try to automatize Twitter and Dev.to before hashnode ig. :)

**Reply by darkterminal** [reply]
But if SCP publish raw story from our member, and what if our member have their own account in DEV/Hashnode/Medium? We will only mention them in SCP content, not on their own account that related to SCP.
